puppet/spec/unit/reports/puppetdb_spec.rb

```diff
@@ -367,7 +366,75 @@ def without_producer_timestamp(json_body)
         end
 
       end
+    end
+  end
+
+  context '#resource_status_to_hash' do
+    before do
+      subject.host = 'localhost'
+    end
+
+    context 'large resources' do
+      let(:huge_title) { (0...2501).map { ('a'..'z').to_a[rand(26)] }.join }
+      let(:large_resource) do
+        stub('resource',
+            { pathbuilder: ['foo', 'bar', 'baz'],
+              path: 'foo',
+              file: 'foo',
+              line: 1,
+              tags: [],
+              time: DateTime.now,
+              skipped: false,
+              containment_path: '\\',
+              provider: 'foo',
+              events: [],
+              type: 'File',
+              resource_type: 'File',
+              title: huge_title,
+              merge_into: nil })
+      end
+
+      let (:status) do
+        Puppet::Resource::Status.new(large_resource)
+      end
+
+      it 'large resources will warn about the index size' do
+        Puppet.expects(:warning).with("resource_event on localhost is very large. Values (resource timestamp + title + type) \
+are 2513 bytes and the hard limit is 2704. Caused by Resource type: \"File\" with title: \"#{huge_title}\" in foo:1")
+        subject.add_resource_status(status)
+        subject.resource_status_to_hash(large_resource)
+      end
+    end
+    context "oversized_resource" do
+      let(:huge_title) { (0...2704).map { ('a'..'z').to_a[rand(26)] }.join }
+      let(:oversized_resource) do
+        stub('resource',
+          { pathbuilder: ['foo', 'bar', 'baz'],
+            path: 'foo',
+            file: 'foo',
+            line: 1,
+            tags: [],
+            provider: 'foo',
+            skipped: false,
+            containment_path: '\\',
+            events: [],
+            time: DateTime.now,
+            type: 'File',
+            resource_type: 'File',
+            title: huge_title,
+            merge_into: nil })
+      end
```

Comment:
can't we move this let in the parent context #resource_status_to_hash?

Reply:
Yes, it cleaned up the code a bit. Thanks.